### DISCUSSION TITLE
Specify Out-of-Memory in Autograder Debug Output

### DISCUSSION THREAD

**Discussion by cohnt** [gradescope/autograder_samples#107] [Ideas]
Right now, if the docker instance runs out of memory, it just kills the program that was taking up that memory. The only output is that the program was killed (with no other information) -- I was only able to diagnose that this was the problem when I increased the memory limit in the Gradescope interface (and it ran successfully). It would be helpful if the system could print some sort of warning or error message when this occurs.

**Comment by adam-brucker** [comment]
Gradescope has a roadmap with ideas located here: https://www.gradescope.com/roadmap. I would post this there as users can vote on ideas.

**Reply by cohnt** [reply]
Just posted there.

**Comment by mattrussell2** [comment]
If you're working in Python and use the subprocess module you can use  `resource.setrlimit(resource.RLIMIT_DATA, (kill_limit, kill_limit))` as a `preexec_fn`, with `kill_limit` as the max usable virtual memory (in bytes). This will gracefully kill the running process if it exceeds the memory limits, and not crash the autograder completely.
